### PR TITLE
Change validation and billing payload

### PR DESCRIPTION
**Pull request by gtuk**
# Pull Request

### Description
The previous implementation was not working for real codes, because inside the xml request the "IK_Abrechnungsstelle" was used as the IK number of the insurance. The IK_Abrechnungsstelle should only be used for the encryption process and inside the request itself it needs to be the "Kostentraegerkennung". It was tested against the AOK with a real code

### CODE REVIEW

**Comment by fongie** [comment]
Hi @gtuk thank you very much for the pull request! I will review it right away. Are you using the client in production right now?

File: src/main/java/com/alextherapeutics/diga/model/AbstractDigaInsuranceInformation.java [hunk 1]
```diff
@@ -33,6 +33,16 @@
      */
     @NonNull
     private String insuranceCompanyIKNumber;
+    /**
+     * The invoice recipients IK number of the insurance that generated this code.
+     */
+    @NonNull
+    private String invoiceRecipientIKNumber;
```
Comment (fongie) [inline src/main/java/com/alextherapeutics/diga/model/AbstractDigaInsuranceInformation.java:40]: this field doesnt seem to be used anywhere? We are using the "IK_des_Rechnungsempfaengers" XML value in invoicing already as the "buyerCompanyCreditorIk" and putting that as the creditor in invoices. I don't think we need it for code validation.

**Review by fongie — CHANGES_REQUESTED** [review]
Thank you for your PR and I am super excited that a real code has been validated! (.. even though you had to submit a fix to do it hehe). 

I think the new field on `AbstractDigaInsuranceInformation` called `invoiceRecipientIKNumber` is not necessary, as mentioned in the review. Please remove it unless I am missing something and I will merge and release a new version straight away :)

File: src/main/java/com/alextherapeutics/diga/model/AbstractDigaInsuranceInformation.java (hunk 1 above)
Comment (gtuk) [reply src/main/java/com/alextherapeutics/diga/model/AbstractDigaInsuranceInformation.java]: True it's not needed at the moment. I removed it

**Comment by gtuk** [comment]
@

> Hi @gtuk thank you very much for the pull request! I will review it right away. Are you using the client in production right now?

Yes we are currently using the lib in production, but only had the change to validate against real codes over the weekend.
So far we were also only able to verify the validation process with a real code and not the api billing, but as soon as we know that the api billing is working i will let you know

**Review by fongie — APPROVED** [review]

**Comment by gtuk** [comment]
Thanks to our library it was a fairly easy process. I will definitely let you know once we tested the validation and the billing against more real insurance codes. Btw. does the release process also upload the new version to the central maven repo?

**Comment by fongie** [comment]
Yes, it does! When we create a new "release" here on github, we run a workflow to release to github packages and to maven central. It should be up in a few minutes on Maven, but sometimes they are slower.  Also, if you are updating very soon (like, today), you need to bypass the local maven cache by using `mvn -U` flag - I always forget that and think the release didn't work. The easiest way of finding out if the update is live is to set the client version to `1.1.1` on your development machine and run `mvn clean install -U` on your project. Not sure how this works if you are using gradle :)

**Comment by fongie** [comment]
Seems to be live already actually on maven, so just update to 1.1.1 and run with -U and you've got it

**Comment by gtuk** [comment]
> Seems to be live already actually on maven, so just update to 1.1.1 and run with -U and you've got it

Perfect, just saw is as well. Thanks again for the lib and for your super fast replies

**Pull request merged**